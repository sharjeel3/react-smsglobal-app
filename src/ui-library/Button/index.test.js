import React from 'react';
import { shallow } from 'enzyme';
import { Button } from './index';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('BUTTON - UI Library', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Button>my button</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be clicked once', () => {
    const mockFn = jest.fn();
    const button = shallow(<Button onClick={mockFn}>Press me</Button>);
    button.simulate('click');
    expect(mockFn.mock.calls.length).toEqual(1);
  });
});
