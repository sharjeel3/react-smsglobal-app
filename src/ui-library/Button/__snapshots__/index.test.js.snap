// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BUTTON - UI Library renders correctly 1`] = `
.c0 {
  font-size: 1em;
  height: 2.5em;
  line-height: 2.5em;
  text-overflow: ellipsis;
  color: #FFFFFF;
  background-color: #22AA4E;
  cursor: pointer;
  padding: 0 2em;
  border-radius: 0.25em;
}

<button
  className="c0"
  onClick={[Function]}
>
  my button
</button>
`;
